### ISSUE TITLE
Handling datasets with multiple targets?

### ISSUE THREAD

**Issue by relf** [rust-ml/linfa#84]
I would like to add the [linnerud dataset](https://scikit-learn.org/stable/datasets/toy_dataset.html#linnerrud-dataset) to linfa-dataset but I've noticed that currently targets are only handled as a one-dimensional array. Should we introduce something like:
```rust
pub type MultiTargetDataset<D, T> =
    DatasetBase<ArrayBase<OwnedRepr<D>, Ix2>, ArrayBase<OwnedRepr<T>, Ix2>>;
```
or is there a better way?

**Comment by bytesnake** [comment]
Hey,
this wasn't considered for now but should be as multi-target prediction is very common.

 The targets matrix can be 1, 2 or three dimensional in three different cases:
 * 1dim - 1 target per N observations
 * 2dim - M targets per N observations
 * 3dim - L probabilities for M targets per N observations

here is L the number of labels, M the number of targets and N the number of points. Because `ndarray` supports three dimensional arrays we should switch the `Dataset` definition to
```rust
pub type Dataset<D, T> = DatasetBase<ArrayBase<OwnedRepr<D>, Ix2>, ArrayBase<OwnedRepr<T>, Ix3>>;
```
and then update the whole machinery in `linfa/dataset/`.

**Comment by bytesnake** [comment]
I added a point to #70 for this feature

**Comment by relf** [comment]
Ok thanks. I was too conservative and not aware of the third case (do you have an example in mind?) 

>   I added a point to #70 for this feature

Does it mean you plan to work on that in the coming weeks?

**Comment by bytesnake** [comment]
yes playing around with this right now, because it is a prerequisite for #66 I will open a PR once I have something useful

**Comment by bytesnake** [comment]
work started in #88, I think the basic mechanism is in place and I will finish the PR in the next days

**Comment by bytesnake** [comment]
finally merged #88, can you give feedback whether everything works as expected?

**Comment by relf** [comment]
Ok, I will. I will resume my work on linnerud dataset. FYI, I am working on a port of [PlsRegression](https://scikit-learn.org/stable/modules/generated/sklearn.cross_decomposition.PLSRegression.html) which is tested against that dataset. If it goes well, maybe it could land in linfa? Where? A new member `linfa-cross-decomposition` or ...? What do you think?

**Comment by bytesnake** [comment]
sounds good, we should add CCA at a later point as well. If you want to implement the SVD version, you can use `TruncatedSvd` from `ndarray-linalg`

**Comment by relf** [comment]
A suggestion: now that we have several targets, we could have `target_names` as well.

**Comment by bytesnake** [comment]
multi-target datasets were introduced #88 and a sample dataset added in #89. I also added a note in #70 regarding target naming
